tests/foreman/cli/test_rhcloud_inventory.py

```diff
@@ -264,3 +266,46 @@ def test_rhcloud_external_links():
 
     :CaseLevel: System
     """
+
+
+@pytest.mark.tier3
+def test_positive_generate_all_reports_job(target_sat):
+    """Generate all reports job via foreman-rake console:
+
+    :id: a9e4bfdb-6d7c-4f8c-ae57-a81442926dd8
+
+    :Steps:
+        1. Disable the Automatic Inventory upload setting.
+        2. Execute Foreman GenerateAllReportsJob via foreman-rake.
+
+    :expectedresults: Reports generation works as expected.
+
+    :BZ: 2110163
+
+    :customerscenario: true
+
+    :CaseAutomation: Automated
+
+    :CaseLevel: System
+    """
+    target_sat.update_setting('allow_auto_inventory_upload', False)
```

Comment:
since you are changing the default, you might want to change it back after the test gets executed, to prevent some interference with other tests